RavenDB-7725

ayende opened the pull request:
- Update to a custom build of ClrMd that allows use to do passive attach.
- On Linux, start the Raven.Debug process, but ask it to wait. Then grant it debug permissions on ourselves, let it run, then remove the permissions.
- No need for any additional security configurations

ayende committed:
RavenDB-7725
- Update to a custom build of ClrMd that allows use to do passive attach.
- On Linux, start the Raven.Debug process, but ask it to wait. Then grant it debug permissions on ourselves, let it run, then remove the permissions.
- No need for any additional security configurations

ppekrol commented on src/Raven.Server/Documents/Handlers/Debugging/ThreadsHandler.cs:
```diff
@@ -19,12 +21,6 @@ public class ThreadsHandler : RequestHandler
         [RavenAction("/admin/debug/threads/stack-trace", "GET", AuthorizationStatus.Operator)]
         public Task StackTrace()
         {
-            if (PlatformDetails.RunningOnPosix)
```
I think that we need to keep it, but for MacOS?

ayende replied:
Yes

ppekrol commented on src/Raven.Server/Raven.Server.csproj:
```diff
@@ -102,6 +102,7 @@
     <PackageReference Include="Microsoft.AspNetCore.WebSockets" Version="2.2.1" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.1.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="3.1.0" />
+    <PackageReference Include="Microsoft.Diagnostics.Runtime-ravendb" Version="1.0.7" />
```
I think we can remove this?

ayende replied:
Yes

ppekrol reviewed (commented):
@ayende did you made any changes to clrmd or just compiled master branch?

ppekrol commented on tools/Raven.Debug/CommandLineApp.cs:
```diff
@@ -139,10 +144,20 @@ private static int ExitWithError(string errMsg, CommandLineApplication cmd)
             var threadInfoList = new List<ThreadInfo>();
 
-            using (var dataTarget = DataTarget.AttachToProcess(processId, attachTimeout))
+            using (var dataTarget = DataTarget.AttachToProcess(processId, attachTimeout, AttachFlag.Passive))
```
Do we want always to attach passively? IIRC it is less accurate, on Windows we could stay with what we had?

ayende replied:
The speed we are seeing in generating stack traces is _huge_, even if it isn't as accurate, that is far more useful

ayende commented:
> @ayende did you made any changes to clrmd or just compiled master branch?

I had to remove a bit of debug code, this is available here: https://github.com/ayende/clrmd

ayende committed:
RavenDB-13702
- Thread stacks view should also include thread _names_, as well as their ids
- PR comments
